org.eclipse.scanning.test/src/org/eclipse/scanning/test/command/ScanRequestCreationTest.java

```diff
@@ -189,6 +190,42 @@ public void testGridCommandWithROINoDetector() throws Exception {
 	}
 
 	@Test
+	public void testGridWithPolygonROI() throws GeneratorException {
+		pi.exec("sr =                             "
+			+	"scan_request(                    "
+			+	"    grid(                        "
+			+	"        axes=(my_scannable, 'y'),"  // Can use Scannable objects or strings.
+			+	"        start=(0, 2),            "
+			+	"        stop=(10, 11),           "
+			+	"        count=(5, 6),            "
+			+	"        roi=poly((0, 0),(1, 5),(1, -2))"
+			+	"    ),                           "
+			+	")                                ");
+		@SuppressWarnings("unchecked")
+		ScanRequest<IROI> request = pi.get("sr", ScanRequest.class);
+
+		Collection<Object> models = request.getCompoundModel().getModels();
+		assertEquals(1, models.size());  // I.e. this is not a compound scan.
+
+		Object model = models.iterator().next();
+
+		Collection<IROI> regions = service.findRegions((GridModel) model, request.getCompoundModel().getRegions());
+		assertEquals(1, regions.size());
+
+		IROI region = regions.iterator().next();
+		assertEquals(PolygonalROI.class, region.getClass());
+
+		PolygonalROI poly = (PolygonalROI) region;
+		assertEquals(3, poly.getNumberOfPoints());
+
+		assertEquals(0., poly.getPoint(0).getPointX(), 1e-8);
+		assertEquals(0., poly.getPoint(0).getPointY(), 1e-8);
+		assertEquals(1., poly.getPoint(1).getPointX(), 1e-8);
+		assertEquals(5., poly.getPoint(1).getPointY(), 1e-8);
+		assertEquals(1., poly.getPoint(2).getPointX(), 1e-8);
+		assertEquals(-2., poly.getPoint(2).getPointY(), 1e-8);
+	}
+	@Test
 	public void testStepCommandWithMonitors() throws Exception {
 		pi.exec("sr =                               "
 			+	"scan_request(                      "
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "scan_request(                      " 4 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)